lib/resty/websocket/client.lua

```diff
@@ -161,6 +168,12 @@ function _M.connect(self, uri, opts)
         return 1
     end
 
+    local custom_headers
+    if headers then
+        custom_headers = concat(headers, "\r\n")
+        custom_headers = "\r\n" .. custom_headers
```

Comment:
We have 2 string concatenations here, which is unfortunate. Hopefully we can avoid creating the temporary Lua string in `custom_header` variable on line 173.

Reply:
@agentzh just rebased; what would you advise we do here?

Reply:
@salazar Seems like we'll have to introduce a Lua table to serve as a string buffer to avoid such unnecessary temporary Lua string creations on line 173 and line 174, and only do `string.concat()` for only once on line 187 when constructing the whole request data string. Oh well. Maybe we could do such optimizations automatically in the future with the wirelang compiler.

Reply:
@agentzh Introducing a new table to serve as a string buffer for this particular case seems a bit... extreme? Another approach I thought was to add an empty string as the first element of `headers` -  e.g., `headers[#headers + 1] = headers[1]; headers[1] = ""`; then, the result of `concat` would already begin with `\r\n`.

Reply:
@salazar That table itself can be recycled. Creating several Lua string GC objects on hot Lua code paths is "extreme" for performance :) The difference might be big, actually. Your proposed way changes the order of the request headers, which may not be desired for some (unusual) use cases. And still, you create one temporary Lua string in that branch though it would not be two temporary Lua string creations.

Reply:
@salazar Well, as I said, such optimizations are relatively big changes to the existing code base. We'd better postpone that for future automatic code generation approaches like wirelang.